sdk/src/types/block/core/wrapper.rs

```diff
@@ -0,0 +1,348 @@
+        self.block.as_validation()
+    }
+}
+
+impl Packable for BlockWrapper {
+    type UnpackError = Error;
+    type UnpackVisitor = ProtocolParameters;
+
+    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
+        self.pack_header(packer)?;
+        self.block.pack(packer)?;
+        self.signature.pack(packer)?;
+
+        Ok(())
+    }
+
+    fn unpack<U: Unpacker, const VERIFY: bool>(
+        unpacker: &mut U,
+        protocol_params: &Self::UnpackVisitor,
+    ) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
+        let start_opt = unpacker.read_bytes();
+
+        let protocol_version = u8::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        if VERIFY && protocol_version != protocol_params.version() {
+            return Err(UnpackError::Packable(Error::ProtocolVersionMismatch {
+                expected: protocol_params.version(),
+                actual: protocol_version,
+            }));
+        }
+
+        let network_id = u64::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        if VERIFY && network_id != protocol_params.network_id() {
+            return Err(UnpackError::Packable(Error::NetworkIdMismatch {
+                expected: protocol_params.network_id(),
+                actual: network_id,
+            }));
+        }
+
+        let issuing_time = u64::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        let slot_commitment_id = SlotCommitmentId::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        let latest_finalized_slot = SlotIndex::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        let issuer_id = IssuerId::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        let block = Block::unpack::<_, VERIFY>(unpacker, protocol_params)?;
+
+        let signature = Signature::unpack::<_, VERIFY>(unpacker, &())?;
+
+        if VERIFY {
+            let block_len = if let (Some(start), Some(end)) = (start_opt, unpacker.read_bytes()) {
+                end - start
+            } else {
+                block.packed_len()
+            };
+
+            if block_len > Self::LENGTH_MAX {
+                return Err(UnpackError::Packable(Error::InvalidBlockLength(block_len)));
+            }
+        }
+
+        let block_wrapper = Self {
+            protocol_params: protocol_params.clone(),
+            issuing_time,
+            slot_commitment_id,
+            latest_finalized_slot,
+            issuer_id,
+            block,
+            signature,
+        };
+
+        Ok(block_wrapper)
+    }
+}
+
+#[cfg(feature = "serde")]
+pub(crate) mod dto {
+    use serde::{Deserialize, Serialize};
+
+    use super::*;
+    use crate::{
+        types::{block::core::dto::BlockDto, TryFromDto},
+        utils::serde::string,
+    };
+
+    /// The block object that nodes gossip around in the network.
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct BlockWrapperDto {
```

Comment:
This is just public and not used for serialization because of the protocol parameters in the BlockWrapper?
Maybe still worth to use it for serialize then?

Reply:
Hmm honestly I don't know. I feel like it would be a bit inconsistent to have only one side 🤷🏻‍♂️

Reply:
Yeah probably fine for now, most of the time it's only used internally and bindings Rust part